src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,367 @@
+                    throw Host.ExceptSchemaMismatch(nameof(schema), "input", _parent.FeatureColumn);
+                _featureColumnType = schema[_featureColumnIndex].Type;
+                if (_featureColumnType.ItemType != NumberType.R4 || !_featureColumnType.IsVector)
+                    throw Host.ExceptUserArg(nameof(schema), "Column '{0}' does not have compatible type. Expected type is vector of float.", _parent.FeatureColumn);
+
+                if (InputSchema[_featureColumnIndex].HasSlotNames(_featureColumnType.VectorSize))
+                    InputSchema[_featureColumnIndex].Metadata.GetValue(MetadataUtils.Kinds.SlotNames, ref _slotNames);
+                else
+                    _slotNames = VBufferUtils.CreateEmpty<ReadOnlyMemory<char>>(_featureColumnType.VectorSize);
+            }
+
+            /// <summary>
+            /// Returns the input columns needed for the requested output columns.
+            /// </summary>
+            private protected override Func<int, bool> GetDependenciesCore(Func<int, bool> activeOutput)
+            {
+                var active = new bool[InputSchema.Count];
+                InputSchema.TryGetColumnIndex(_parent.FeatureColumn, out int featureCol);
+                active[featureCol] = true;
+                return col => active[col];
+            }
+
+            public override void Save(ModelSaveContext ctx)
+                => _parent.Save(ctx);
+
+            // The FeatureContributionCalculatingTransformer produces two sets of columns: the columns obtained from scoring and the FeatureContribution column.
+            // If the argument stringify is true, the type of the FeatureContribution column is string, otherwise it is a vector of float.
+            protected override Schema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                // Add FeatureContributions column.
+                var builder = new MetadataBuilder();
+                builder.Add(InputSchema[_featureColumnIndex].Metadata, x => x == MetadataUtils.Kinds.SlotNames);
+                return new[] { new Schema.DetachedColumn(DefaultColumnNames.FeatureContributions, new VectorType(NumberType.R4, _featureColumnType.ValueCount), builder.GetMetadata()) };
+            }
+
+            protected override Delegate MakeGetter(Row input, int iinfo, Func<int, bool> active, out Action disposer)
+            {
+                disposer = null;
+                if (active(iinfo))
+                    return GetContributionGetter(input, _featureColumnIndex);
+                return null;
+            }
+
+            public Delegate GetContributionGetter(Row input, int colSrc)
+            {
+                Contracts.CheckValue(input, nameof(input));
+                Contracts.Check(0 <= colSrc && colSrc < input.Schema.Count);
+
+                var typeSrc = input.Schema[colSrc].Type;
+                Func<Row, int, ValueGetter<VBuffer<float>>> del = GetValueGetter<int>;
```

Comment:
>GetValueGetter [](start = 66, length = 14)

I suggest changing `GetValueGetter` to return `Delegate`. Then you can replace lines 262-263 with `Utils.MarshalInvoke(del, input , colSrc)`